src/giskardpy/plugin_world_visualization.py

```diff
@@ -0,0 +1,62 @@
+import hashlib
+
+import py_trees
+import rospy
+from visualization_msgs.msg import Marker, MarkerArray
+
+from plugin import GiskardBehavior
+import giskardpy.identifier as identifier
+from giskardpy.tfwrapper import lookup_pose, get_full_frame_name
+
+
+class WorldVisualizationBehavior(GiskardBehavior):
+    def __init__(self, name, ensure_publish=False):
+        super(WorldVisualizationBehavior, self).__init__(name)
+        self.map_frame = self.get_god_map().get_data(identifier.map_frame)
+        self.marker_namespace = u'planning_world_visualization'
+        self.ensure_publish = ensure_publish
+
+    def setup(self, timeout):
+        self.publisher = rospy.Publisher(u'~visualization_marker_array', MarkerArray, queue_size=1)
+        self.ids = set()
+        return super(WorldVisualizationBehavior, self).setup(timeout)
+
+    def update(self):
+        markers = []
+        objects = self.get_world().get_objects().values()
```

Comment:
This won't work in python 3, because `values` doesn't return a list anymore. Iterate over it in directly in the loop

Reply:
ah no, it will work, because you only use `objects` in the loop. But you should still do the change because it's cleaner.